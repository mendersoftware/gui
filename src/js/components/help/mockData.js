// Copyright 2020 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {
    hasDeviceConfig: true,
    hasMonitor: true
  },
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '2003-01-01'
  }
};
