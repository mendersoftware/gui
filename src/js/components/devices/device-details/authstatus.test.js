import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import AuthStatus from './authstatus';
import { defaultState, undefineds } from '../../../../../tests/mockData';
const mockStore = configureStore([thunk]);

describe('AuthStatus Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({ ...defaultState });
  });

  it('renders correctly', async () => {
    const { baseElement } = render(
      <Provider store={store}>
        <AuthStatus
          device={{
            ...defaultState.devices.byId.a1,

            auth_sets: [
              ...defaultState.devices.byId.a1.auth_sets,
              {
                ...defaultState.devices.byId.a1.auth_sets[0],
                status: 'pending'
              }
            ]
          }}
          toggleAuthsets={jest.fn}
        />
      </Provider>
    );
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
