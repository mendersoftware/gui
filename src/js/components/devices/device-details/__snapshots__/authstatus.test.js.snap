// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthStatus Component renders correctly 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.3929rem;
}

.emotion-0 iconButton {
  margin-right: 8px;
}

.emotion-1 {
  max-width: 100%;
  font-family: Lato,sans-serif;
  font-size: 0.7545rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 24px;
  color: #fff;
  background-color: #337a87;
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: unset;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
}

.emotion-1.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-1 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #bcbcbc;
  font-size: 0.6964rem;
}

.emotion-1 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: rgb(35, 85, 94);
}

.emotion-1 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: rgb(65, 10, 46);
}

.emotion-1 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.5804rem;
}

.emotion-1 .MuiChip-icon {
  margin-left: 4px;
  margin-right: -4px;
  font-size: 18px;
  color: inherit;
}

.emotion-1 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(255, 255, 255, 0.7);
  font-size: 16px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
  margin-right: 4px;
  margin-left: -4px;
}

.emotion-1 .MuiChip-deleteIcon:hover {
  color: rgba(10, 10, 11, 0.4);
}

.emotion-1 .MuiChip-deleteIcon:hover,
.emotion-1 .MuiChip-deleteIcon:active {
  color: #fff;
}

.emotion-2 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 8px;
  padding-right: 8px;
  white-space: nowrap;
}

.emotion-3 {
  background-color: #f7f7f7;
  border-color: #e9e9e9;
  border-style: solid;
  border-width: 1px;
  margin-bottom: 16px;
  min-width: 700px;
  padding: 16px;
}

.emotion-4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.3929rem;
  color: #337a87;
}

.emotion-4 iconButton {
  margin-right: 8px;
}

.emotion-4.read {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-5 {
  position: absolute;
  top: -5px;
  bottom: 0;
  left: -5px;
  right: -5px;
  border: 1px dashed #337a87;
  border-radius: 50%;
}

.emotion-5.read {
  border-color: rgba(0, 0, 0, 0.38);
}

.emotion-6 .header,
.emotion-6 .MuiAccordion-root {
  border-bottom: 1px solid #cfcfcf;
}

.emotion-6 .columnHeader,
.emotion-6 .MuiAccordionSummary-root,
.emotion-6 .MuiAccordionSummary-content {
  cursor: default;
}

.emotion-6 .header,
.emotion-6 .body .MuiAccordionSummary-content {
  display: grid;
  grid-column-gap: 16px;
  grid-template-columns: 0.5fr 1fr 2fr 2fr 2fr;
}

.emotion-7 {
  padding: 16px;
}

.emotion-7.columns-4 {
  grid-template-columns: 0.5fr 1fr 2fr 2fr;
}

.emotion-8 {
  background-color: #fff;
  color: rgba(10, 10, 11, 0.78);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  position: relative;
  -webkit-transition: margin 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: margin 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  overflow-anchor: none;
  border: none;
  box-shadow: none;
  padding: 0;
  background-color: #d4e9e7;
}

.emotion-8::before {
  position: absolute;
  left: 0;
  top: -1px;
  right: 0;
  height: 1px;
  content: "";
  opacity: 1;
  background-color: rgba(0, 0, 0, 0.12);
  -webkit-transition: opacity 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-8:first-of-type::before {
  display: none;
}

.emotion-8.Mui-expanded::before {
  opacity: 0;
}

.emotion-8.Mui-expanded:first-of-type {
  margin-top: 0;
}

.emotion-8.Mui-expanded:last-of-type {
  margin-bottom: 0;
}

.emotion-8.Mui-expanded+.emotion-8.Mui-expanded::before {
  display: none;
}

.emotion-8.Mui-disabled {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-8.Mui-expanded {
  margin: 16px 0;
}

.emotion-8:before {
  display: none;
}

.emotion-8.Mui-expanded {
  margin: auto;
  background-color: #f7f7f7;
}

.emotion-8:before {
  display: none;
}

.emotion-8$expanded {
  margin: auto;
}

.emotion-8 .columns-4 .MuiAccordionSummary-content {
  grid-template-columns: 0.5fr 1fr 2fr 2fr;
}

.emotion-8 .MuiAccordionDetails-root {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 48px;
  padding: 0px 16px;
  -webkit-transition: min-height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: min-height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  margin-bottom: 0;
  height: 48px;
}

.emotion-9::-moz-focus-inner {
  border-style: none;
}

.emotion-9.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-9 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-9.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-9.Mui-disabled {
  opacity: 0.38;
}

.emotion-9:hover:not(.Mui-disabled) {
  cursor: pointer;
}

.emotion-9.Mui-expanded {
  min-height: 64px;
}

.emotion-9.Mui-expanded {
  height: 48px;
  min-height: 48px;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 12px 0;
  -webkit-transition: margin 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: margin 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-10.Mui-expanded {
  margin: 20px 0;
}

.emotion-10.Mui-expanded {
  margin: 0;
}

.emotion-10>:last-child {
  padding-right: 12px;
}

.emotion-13 {
  height: 0;
  overflow: hidden;
  -webkit-transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  visibility: hidden;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.emotion-15 {
  width: 100%;
}

.emotion-16 {
  padding: 8px 16px 16px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-24 {
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  margin-top: 16px;
}

.emotion-25 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Lato,sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #5d0f43;
  border-radius: 2px;
  font-size: 14px;
  font-weight: bold;
  padding: 10px 15px;
}

.emotion-25::-moz-focus-inner {
  border-style: none;
}

.emotion-25.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-25 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-25:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(93, 15, 67, 0.04);
}

@media (hover: none) {
  .emotion-25:hover {
    background-color: transparent;
  }
}

.emotion-25.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-25:hover {
  colors: #337a87;
}

.emotion-25.MuiButton-text {
  color: rgba(10, 10, 11, 0.78);
}

.emotion-26 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div
  class="margin-bottom "
>
  <div
    class="flexbox space-between center-aligned"
  >
    <div
      class="flexbox center-aligned"
    >
      <h4>
        Authentication status
      </h4>
      <div
        class="flexbox center-aligned margin-left margin-right"
      >
        <div
          class="capitalized"
        >
          accepted
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium green emotion-0"
          data-testid="CheckCircleIcon"
          focusable="false"
          style="margin: 12px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
          />
        </svg>
      </div>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary emotion-1"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall emotion-2"
        >
          new request
        </span>
      </div>
    </div>
    <div
      class="flexbox centered"
    />
  </div>
  <div />
  <div
    class="emotion-3"
  >
    <div
      class="margin-bottom-small flexbox space-between"
    >
      Authorization sets
      <div
        class=""
        data-mui-internal-clone-element="true"
      >
        <div
          class="relative"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium  emotion-4"
            data-testid="HelpIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 17h-2v-2h2zm2.07-7.75-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25"
            />
          </svg>
          <div
            class="emotion-5 "
          />
        </div>
      </div>
    </div>
    <div
      class="authsets emotion-6"
    >
      <div
        class="header columns-5 emotion-7"
      >
        <div
          class="columnHeader"
        />
        <div
          class="columnHeader"
        >
          Status
        </div>
        <div
          class="columnHeader"
        >
          Public key
        </div>
        <div
          class="columnHeader"
        >
          Time of request
        </div>
        <div
          class="columnHeader"
        >
          Actions
        </div>
      </div>
      <div
        class="body relative"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 MuiAccordion-root MuiAccordion-gutters emotion-8"
        >
          <div
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters columns-5 emotion-9"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters emotion-10"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary emotion-1"
                style="justify-self: flex-start;"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall emotion-2"
                >
                  new
                </span>
              </div>
              <div
                class="capitalized"
              >
                pending
              </div>
              <a>
                show key
              </a>
              <time
                datetime="2019-01-13T06:25:00+00:00"
              >
                2019-01-13 06:25
              </time>
              <div
                class="action-buttons flexbox"
              >
                <a>
                  Accept
                </a>
                <a>
                  Reject
                </a>
                <a>
                  Dismiss
                </a>
              </div>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-13"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical emotion-14"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-15"
              >
                <div
                  class="MuiAccordion-region"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root emotion-16"
                  >
                    <div
                      style="flex-grow: 1;"
                    />
                    <p
                      class="bold expanded"
                    >
                       Are you sure you want to continue?
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 MuiAccordion-root MuiAccordion-gutters emotion-8"
        >
          <div
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters columns-5 emotion-9"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters emotion-10"
            >
              <div
                class="capitalized"
              >
                Active
              </div>
              <div
                class="capitalized"
              >
                accepted
              </div>
              <a>
                show key
              </a>
              <time
                datetime="2019-01-13T06:25:00+00:00"
              >
                2019-01-13 06:25
              </time>
              <div
                class="action-buttons flexbox"
              >
                <div>
                  Accept
                </div>
                <a>
                  Reject
                </a>
                <a>
                  Dismiss
                </a>
              </div>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-13"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical emotion-14"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-15"
              >
                <div
                  class="MuiAccordion-region"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root emotion-16"
                  >
                    <div
                      style="flex-grow: 1;"
                    />
                    <p
                      class="bold expanded"
                    >
                       Are you sure you want to continue?
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flexbox emotion-24"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary emotion-25"
        tabindex="0"
        type="button"
      >
        Decommission device
        <span
          class="MuiTouchRipple-root emotion-26"
        />
      </button>
    </div>
  </div>
</div>
`;
